File: Application/src/main/java/com/shareyourproxy/api/service/UserContactService.java
package com.shareyourproxy.api.service;

import com.shareyourproxy.api.domain.model.User;

import retrofit.http.Body;
import retrofit.http.DELETE;
import retrofit.http.PUT;
import retrofit.http.Path;
import rx.Observable;

/**
 * Pokemon that a user wants to collect.
 */
public interface UserContactService {
    /**
     * add a User contact id.
     *
     * @param userId unique id for {@link User} table
     */
    @PUT("/users/{userId}/contacts.json")
    Observable<String> addUserContact(
        @Path("userId") String userId, @Body String contactId);

    @DELETE("/users/{userId}/contacts/{contactId}.json")
    Observable<String> deleteUserContact(
        @Path("userId") String userId, @Path("contactId") String contactId);

}

File: Application/src/main/kotlin/com/shareyourproxy/api/service/UserContactService.kt
package com.shareyourproxy.api.service

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.PUT
import retrofit2.http.Path
import rx.Observable

/**
 * Pokemon that a user wants to collect.
 */
interface UserContactService {
    /**
     * add a User contact id.
     * @param userId unique id for [User] table
     */
    @PUT("/users/{userId}/contacts.json")
    fun addUserContact(@Path("userId") userId: String, @Body contactId: String): Observable<String>

    @DELETE("/users/{userId}/contacts/{contactId}.json")
    fun deleteUserContact(@Path("userId") userId: String, @Path("contactId") contactId: String): Observable<String>

}
